docs/plugins/queue/mmap/README.md

```diff
@@ -0,0 +1,66 @@
+# Design
+The mmap-queue is a big, fast and persistent queue based on the memory mapped files. One mmap-queue has a directory to store the whole data. The Queue directory is made up with many segments and 1 meta file, which references the design of the [bigqueue](https://github.com/grandecola/bigqueue). 
```

Comment:
```suggestion
The mmap-queue is a big, fast, and persistent queue based on the memory-mapped files. One mmap-queue has a directory to store the whole data. The queue directory is made up of many segments and 1 metafile. This is originally implemented by [bigqueue](https://github.com/grandecola/bigqueue) project, we changed it a little for fitting the Satellite project requirements.
```

Reply:
Please note, `reference` usually means taking the design idea, but here, we copied some of it.